package/MDAnalysis/coordinates/CHEMFILES.py

```diff
@@ -0,0 +1,205 @@
+#
+"""
+TODO
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import numpy as np
+
+from . import base, core
+
+from chemfiles import Trajectory, Frame, Atom, UnitCell, Residue
+
+class ChemfilesReader(base.ReaderBase):
+    """
+    TODO
+    """
+    format = 'CHEMFILES'
+    units = {'time': 'fs', 'length': 'Angstrom'}
+
+    def __init__(self, filename, format="", **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str
+            trajectory filename
+        format : str (optional)
+            use the given format name instead of guessing from the extension
+            TODO: link to the doc
+        **kwargs : dict
+            General reader arguments.
+        """
+        super(ChemfilesReader, self).__init__(filename, **kwargs)
+        self._format = format
+        self._cached_n_atoms = None
+        self._open()
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        self.next()
+
+    def _open(self):
+        self._file = Trajectory(self.filename, 'r', self._format)
+        self._closed = False
+        self._step = 0
+
+    def close(self):
+        """close reader"""
+        if not self._closed:
+            self._file.close()
+            self._closed = True
+
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        return self._file.nsteps()
+
+    @property
+    def n_atoms(self):
+        """number of atoms in the first frame of the trajectory"""
+        if self._cached_n_atoms is None:
+            self._cached_n_atoms = self._file.read_step(0).natoms()
+        return self._cached_n_atoms
+
+
+    def _reopen(self):
+        """reopen trajectory"""
+        self.close()
+        self._open()
+
+    def _read_frame(self, i):
+        """read frame i"""
+        self._step = i
+        return self._read_next_timestep()
+
+    def _read_next_timestep(self, ts=None):
+        """copy next frame into timestep"""
+        if self._step >= self.n_frames:
+            raise IOError('trying to go over trajectory limit')
+        if ts is None:
+            ts = self.ts
+        self.ts = ts
+        frame = self._file.read_step(self._step)
+        self._frame_to_ts(frame, ts)
+        ts.frame = self._step
+        self._step += 1
+        return ts
+
+    def _frame_to_ts(self, frame, ts):
+        """convert a chemfiles frame to a :class:`TimeStep`"""
+        if frame.natoms() != self.n_atoms:
+            raise IOError(
+                "The number of atom changed in the trajectory. This is not " +
+                "supported by MDAnalysis."
+            )
+
+        cell = frame.cell()
+        dimensions = cell.lengths() + cell.angles()
+        ts.dimensions = np.array(dimensions, dtype=np.float32)
+
+        # TODO: should we make sure to keep the frame alive / copy the data?
+        # TODO: docs mention FORTRAN order, chemfiles uses C order
```

Comment:
This seems to work well, is there an issue with the doc or is the conversion automatically done for me?

Reply:
No like this you assign a new reference. If you would to `ts.positions[:]` then python would to an inplace update. You are just doing a reference update.

Reply:
the inplace update should also do the type conversion automatically.